src/java.base/share/classes/jdk/internal/foreign/abi/fallback/FallbackLinker.java

```diff
@@ -283,4 +293,44 @@ private static Object readValue(MemorySegment seg, MemoryLayout layout) {
         assert layout == null;
         return null;
     }
+
+    @Override
+    public Map<String, MemoryLayout> canonicalLayouts() {
+        return CANONICAL_LAYOUTS;
+    }
+
+    static final Map<String, MemoryLayout> CANONICAL_LAYOUTS;
+
+    static {
+        int wchar_size = LibFallback.wcharSize();
+        MemoryLayout wchartLayout = switch(wchar_size) {
+            case 2 -> JAVA_CHAR; // prefer JAVA_CHAR
+            default -> FFIType.layoutFor(wchar_size);
+        };
+
+        CANONICAL_LAYOUTS = Map.ofEntries(
```

Comment:
`LibFallback::wcharSize()` and other getters for `LibFallback.NativeConstants` fields can throw an error when `LibFallback.SUPPORTED` is `false` due to the `fallbackLinker` library not being present, so this static initializer should be made into a method instead:
```suggestion
    static final Map<String, MemoryLayout> CANONICAL_LAYOUTS = initCanonicalLayouts();

    private static Map<String, MemoryLayout> initCanonicalLayouts() {
        if (!isSupported()) {
            return null;
        }

        int wchar_size = LibFallback.wcharSize();
        MemoryLayout wchartLayout = switch(wchar_size) {
            case 2 -> JAVA_CHAR; // prefer JAVA_CHAR
            default -> FFIType.layoutFor(wchar_size);
        };

        return Map.ofEntries(
```

Reply:
Good catch! I've chosen a slightly different solution though.